pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/model/elastic/migration/CohortTagMigratorTest.java

```diff
@@ -0,0 +1,173 @@
+import org.broadinstitute.dsm.util.ElasticTestUtil;
+import org.broadinstitute.dsm.util.TestParticipantUtil;
+import org.broadinstitute.dsm.util.proxy.jackson.ObjectMapperSingleton;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+@Slf4j
+public class CohortTagMigratorTest extends DbAndElasticBaseTest {
+    private static final DDPInstanceDao ddpInstanceDao = new DDPInstanceDao();
+    private static DDPInstanceDto os1DdpInstanceDto;
+    private static DDPInstanceDto os2DdpInstanceDto;
+    private static String os1InstanceName;
+    private static String os2InstanceName;
+    private static String esIndex;
+    private static int participantCounter;
+    private static final List<ParticipantDto> os1Participants = new ArrayList<>();
+    private static final List<ParticipantDto> os2Participants = new ArrayList<>();
+    private static final String OS1_TAG = "OS";
+    private static final String OS2_TAG = "OS PE-CGS";
+    private static final String baseInstanceName = "cohorttag";
+
+    private enum Cohort {
+        OS1, OS2, OS1_OS2
+    }
+
+    @BeforeClass
+    public static void setup() throws Exception {
+        participantCounter = 1;
+        // NEW_OSTEO_INSTANCE_NAME and OLD_OSTEO_INSTANCE_NAME are already hardcoded in the cohort export code,
+        // so we either need to use them or change the code under test. Since the latter really needs a rewrite
+        // (with tests!) we do it this way for now.
+        // TODO: revisit this -- DC
+        os1InstanceName = OLD_OSTEO_INSTANCE_NAME;
+        os2InstanceName = NEW_OSTEO_INSTANCE_NAME;
+        esIndex = ElasticTestUtil.createIndex(baseInstanceName, "elastic/lmsMappings.json", null);
+        // OS1 realm is not in the test DB, OS2 is
+        os1DdpInstanceDto = DdpInstanceGroupTestUtil.createTestDdpInstance(os1InstanceName, esIndex);
+        os2DdpInstanceDto = ddpInstanceDao.getDDPInstanceByInstanceName(os2InstanceName).orElseThrow();
+    }
+
+    @AfterClass
+    public static void tearDown() {
+        ElasticTestUtil.deleteIndex(esIndex);
+        ddpInstanceDao.delete(os1DdpInstanceDto.getDdpInstanceId());
+    }
+
+    @After
+    public void deleteParticipantData() {
+        os1Participants.forEach(ptp -> {
+            CohortTagTestUtil.deleteTag(ptp.getDdpParticipantIdOrThrow(), OS1_TAG);
+            TestParticipantUtil.deleteParticipant(ptp.getParticipantId().orElseThrow());
+        });
+        os2Participants.forEach(ptp -> {
+            CohortTagTestUtil.deleteTag(ptp.getDdpParticipantIdOrThrow(), OS2_TAG);
+            TestParticipantUtil.deleteParticipant(ptp.getParticipantId().orElseThrow());
+        });
+    }
+
+    @Test
+    public void xxxTest() {
+        // create participants and tags for OS1, OS2 and for both
+        createParticipant(Cohort.OS1);
+        createParticipant(Cohort.OS2);
+        createParticipant(Cohort.OS1_OS2);
+
```

Comment:
I like this approach--just run the export for different os1/os2 in different ways, and verify that after each export, the tags are what you expect them to be.